uber/rides-ios-sdk
# Remove UberButton_DEPRECATED

## Description
This PR removes the last remaining references to UberButton_DEPRECATED and updates RideRequestButton to match the previous UI. The sample app UberSDK has also been updated to allow manually testing the new button.

## Changes

### Remove UberButton_DEPRECATED
This UI was deprecated in an earlier PR and was only referenced in RideRequestButton. After migrating RideRequestButton to subclass the new UberButton, UberButton_DEPRECATED could safely be removed.

### Sample App
Minor changes were made to the sample all to add the new RideRequestButton and make some more space for UI (AuthResponseView).

## Testing
### Unit Tests
Existing unit tests for the deprecated UberButton were updated to support the new button

### Manual Testing
1. Run the sample app
2. Login using a test account in the first section
3. The last section (Request a Ride Button) should automatically update to the following
<img width="404" alt="Screenshot 2024-08-26 at 1 32 59 PM" src="https://github.com/user-attachments/assets/b30fe6e7-a145-4674-95bb-1ae24ca9a675">
